tests/pyfunc/test_warn_dependency_requirement_mismatches.py

```diff
@@ -0,0 +1,118 @@
+import os
+from unittest import mock
+
+import pytest
+import sklearn.datasets
+import sklearn.linear_model
+import sklearn.neighbors
+
+from mlflow.pyfunc import _warn_dependency_requirement_mismatches
+
+
+@pytest.mark.large
+def test_warn_dependency_requirement_mismatches(tmpdir):
+    import cloudpickle
+    import mlflow.utils.requirements_utils
+
+    model_path = tmpdir
+    req_file_path = os.path.join(tmpdir, "requirements.txt")
+    with open(req_file_path, "w") as fp:
+        fp.write(f"cloudpickle=={cloudpickle.__version__}\nscikit-learn=={sklearn.__version__}\n")
```

Comment:
```suggestion
    req_file = tmpdir.join("requirements.txt")
    req_file.write(f"cloudpickle=={cloudpickle.__version__}\nscikit-learn=={sklearn.__version__}\n")
```

Trick to simplify code like this.